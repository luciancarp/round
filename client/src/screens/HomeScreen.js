import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import StyledPage from '../components/layout/StyledPage'
import StyledTitle from '../components/layout/StyledTitle'
import IssuesItem from '../components/IssuesItem'
import Spinner from '../components/common/Spinner'
import StyledUnorderedList from '../components/layout/StyledUnorderedList'

import { getIssues } from '../actions/issuesActions'

class HomeScreen extends Component {
  constructor (props) {
    super(props)
    this.state = {
      title: '',
      issues: [],
      loading: true
    }
  }

  componentDidMount () {
    this.props.getIssues()
  }

  componentDidUpdate (prevProps, prevState) {
    if (prevProps.issues.issues !== this.props.issues.issues) {
      const { issues } = this.props.issues
      this.setState({
        title: 'Issues',
        issues: issues,
        loading: false
      })
    }
  }

  render () {
    const { issues } = this.state
    return (
      <StyledPage>
        {this.state.loading && <Spinner />}
        <StyledTitle>{this.state.title}</StyledTitle>
        <StyledUnorderedList>
          {
            issues.map(issue => <IssuesItem
              key={issue._id}
              id={issue._id}
              name={issue.name}
              description={issue.description}
              date={issue.date}
            />)
          }
        </StyledUnorderedList>
      </StyledPage>
    )
  }
}

HomeScreen.propTypes = {
  auth: PropTypes.object.isRequired,
  issues: PropTypes.object.isRequired,
  getIssues: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
  issues: state.issues,
  auth: state.auth
})

export default connect(
  mapStateToProps,
  { getIssues }
)(HomeScreen)
