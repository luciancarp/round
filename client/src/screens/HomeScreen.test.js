import React from 'react'
import { shallow } from 'enzyme'
import HomeScreen from './HomeScreen'

it('should render correctly with no props', () => {
  const component = shallow(<HomeScreen />)
  expect(component).toMatchSnapshot()
})
