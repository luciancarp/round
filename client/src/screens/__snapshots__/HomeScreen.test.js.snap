// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Home
        </ForwardRef>,
        <p>
          Maecenas lacinia efficitur ante eget facilisis. Maecenas sed varius quam, eget cursus nisi. In sit amet auctor dui, id iaculis elit. Nunc placerat odio ut suscipit lobortis. Nam rutrum mattis eros, a mollis odio egestas ut. Ut tempus eros at ex ullamcorper dapibus. Nunc sed consequat urna. Aliquam enim dolor, vehicula nec diam eget, ornare tincidunt nunc. Nam tempor venenatis tellus id tristique. Sed laoreet diam et molestie venenatis. Integer posuere, ligula et iaculis malesuada, diam nulla pellentesque lorem, non vestibulum purus ligula non felis. In hac habitasse platea dictumst. Cras interdum tempor ultrices. Mauris in tempor nunc. Duis lacinia nunc non nibh porta, eget vulputate purus sagittis. Quisque ultricies pulvinar turpis at dignissim. In volutpat massa at leo maximus, a suscipit ex faucibus. Nullam eu ante in sem lacinia faucibus. Sed non eros eu enim lobortis laoreet.
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Home",
        },
        "ref": null,
        "rendered": "Home",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
  color: ",
              "#333333",
              ";
  font-size: ",
              "32",
              "px;
",
            ],
          },
          "displayName": "styled.h1",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "h1",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Maecenas lacinia efficitur ante eget facilisis. Maecenas sed varius quam, eget cursus nisi. In sit amet auctor dui, id iaculis elit. Nunc placerat odio ut suscipit lobortis. Nam rutrum mattis eros, a mollis odio egestas ut. Ut tempus eros at ex ullamcorper dapibus. Nunc sed consequat urna. Aliquam enim dolor, vehicula nec diam eget, ornare tincidunt nunc. Nam tempor venenatis tellus id tristique. Sed laoreet diam et molestie venenatis. Integer posuere, ligula et iaculis malesuada, diam nulla pellentesque lorem, non vestibulum purus ligula non felis. In hac habitasse platea dictumst. Cras interdum tempor ultrices. Mauris in tempor nunc. Duis lacinia nunc non nibh porta, eget vulputate purus sagittis. Quisque ultricies pulvinar turpis at dignissim. In volutpat massa at leo maximus, a suscipit ex faucibus. Nullam eu ante in sem lacinia faucibus. Sed non eros eu enim lobortis laoreet.",
        },
        "ref": null,
        "rendered": "Maecenas lacinia efficitur ante eget facilisis. Maecenas sed varius quam, eget cursus nisi. In sit amet auctor dui, id iaculis elit. Nunc placerat odio ut suscipit lobortis. Nam rutrum mattis eros, a mollis odio egestas ut. Ut tempus eros at ex ullamcorper dapibus. Nunc sed consequat urna. Aliquam enim dolor, vehicula nec diam eget, ornare tincidunt nunc. Nam tempor venenatis tellus id tristique. Sed laoreet diam et molestie venenatis. Integer posuere, ligula et iaculis malesuada, diam nulla pellentesque lorem, non vestibulum purus ligula non felis. In hac habitasse platea dictumst. Cras interdum tempor ultrices. Mauris in tempor nunc. Duis lacinia nunc non nibh porta, eget vulputate purus sagittis. Quisque ultricies pulvinar turpis at dignissim. In volutpat massa at leo maximus, a suscipit ex faucibus. Nullam eu ante in sem lacinia faucibus. Sed non eros eu enim lobortis laoreet.",
        "type": "p",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  /* margin: auto; */
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 0px;
  padding: ",
          "14",
          "px;
  min-height: calc(100vh - 225px);
  /* text-align: center; */

  @media ",
          "(min-width: 1024px)",
          " {  
    max-width: 1024px;
  }

  @media ",
          "(min-width: 2560px)",
          " {
    max-width: 1400px;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Home
          </ForwardRef>,
          <p>
            Maecenas lacinia efficitur ante eget facilisis. Maecenas sed varius quam, eget cursus nisi. In sit amet auctor dui, id iaculis elit. Nunc placerat odio ut suscipit lobortis. Nam rutrum mattis eros, a mollis odio egestas ut. Ut tempus eros at ex ullamcorper dapibus. Nunc sed consequat urna. Aliquam enim dolor, vehicula nec diam eget, ornare tincidunt nunc. Nam tempor venenatis tellus id tristique. Sed laoreet diam et molestie venenatis. Integer posuere, ligula et iaculis malesuada, diam nulla pellentesque lorem, non vestibulum purus ligula non felis. In hac habitasse platea dictumst. Cras interdum tempor ultrices. Mauris in tempor nunc. Duis lacinia nunc non nibh porta, eget vulputate purus sagittis. Quisque ultricies pulvinar turpis at dignissim. In volutpat massa at leo maximus, a suscipit ex faucibus. Nullam eu ante in sem lacinia faucibus. Sed non eros eu enim lobortis laoreet.
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Home",
          },
          "ref": null,
          "rendered": "Home",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
  color: ",
                "#333333",
                ";
  font-size: ",
                "32",
                "px;
",
              ],
            },
            "displayName": "styled.h1",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "h1",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Maecenas lacinia efficitur ante eget facilisis. Maecenas sed varius quam, eget cursus nisi. In sit amet auctor dui, id iaculis elit. Nunc placerat odio ut suscipit lobortis. Nam rutrum mattis eros, a mollis odio egestas ut. Ut tempus eros at ex ullamcorper dapibus. Nunc sed consequat urna. Aliquam enim dolor, vehicula nec diam eget, ornare tincidunt nunc. Nam tempor venenatis tellus id tristique. Sed laoreet diam et molestie venenatis. Integer posuere, ligula et iaculis malesuada, diam nulla pellentesque lorem, non vestibulum purus ligula non felis. In hac habitasse platea dictumst. Cras interdum tempor ultrices. Mauris in tempor nunc. Duis lacinia nunc non nibh porta, eget vulputate purus sagittis. Quisque ultricies pulvinar turpis at dignissim. In volutpat massa at leo maximus, a suscipit ex faucibus. Nullam eu ante in sem lacinia faucibus. Sed non eros eu enim lobortis laoreet.",
          },
          "ref": null,
          "rendered": "Maecenas lacinia efficitur ante eget facilisis. Maecenas sed varius quam, eget cursus nisi. In sit amet auctor dui, id iaculis elit. Nunc placerat odio ut suscipit lobortis. Nam rutrum mattis eros, a mollis odio egestas ut. Ut tempus eros at ex ullamcorper dapibus. Nunc sed consequat urna. Aliquam enim dolor, vehicula nec diam eget, ornare tincidunt nunc. Nam tempor venenatis tellus id tristique. Sed laoreet diam et molestie venenatis. Integer posuere, ligula et iaculis malesuada, diam nulla pellentesque lorem, non vestibulum purus ligula non felis. In hac habitasse platea dictumst. Cras interdum tempor ultrices. Mauris in tempor nunc. Duis lacinia nunc non nibh porta, eget vulputate purus sagittis. Quisque ultricies pulvinar turpis at dignissim. In volutpat massa at leo maximus, a suscipit ex faucibus. Nullam eu ante in sem lacinia faucibus. Sed non eros eu enim lobortis laoreet.",
          "type": "p",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  /* margin: auto; */
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 0px;
  padding: ",
            "14",
            "px;
  min-height: calc(100vh - 225px);
  /* text-align: center; */

  @media ",
            "(min-width: 1024px)",
            " {  
    max-width: 1024px;
  }

  @media ",
            "(min-width: 2560px)",
            " {
    max-width: 1400px;
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
